apps/src/templates/teacherDashboard/AssignmentSelector.jsx

```diff
@@ -47,23 +54,31 @@ export default class AssignmentSelector extends Component {
     disabled: PropTypes.bool,
   };
 
-  getVersionYears = assignmentFamilyName => {
+  /**
+   * Given an assignment family, return a list of versions representing valid
+   * assignments within that family, with highest year numbers first.
+   * @param {AssignmentFamilyShape} assignmentFamilyName
+   * @returns {Object} Version object with the following properties:
+   *   {string} year The year associated with this version, used as a key for
+   *     identifying this version programmatically.
+   *   {string} title The UI string associated with this version.
+   */
+  getVersions = assignmentFamilyName => {
     if (!assignmentFamilyName) {
       return [];
     }
-    return _.values(this.props.assignments)
+    const versions = _.values(this.props.assignments)
       .filter(assignment => assignment.assignment_family_name === assignmentFamilyName)
-      .map(assignment => assignment.version_year)
-      .sort()
-      .reverse();
+      .map(getVersion);
+    return _.sortBy(versions, 'year').reverse();
```

Comment:
All of the operations here are

Reply:
nice! I wasn't sure lodash could do all those things, but of course it can.